src/core/Akka/Actor/SupervisorStrategy.cs

```diff
@@ -193,10 +190,9 @@ public static SupervisorStrategy DefaultStrategy
     /// </summary>
     public class OneForOneStrategy : SupervisorStrategy
     {
-        private readonly int _maxNumberOfRetries;
-        private readonly int _withinTimeRangeMilliseconds;
-        private readonly Func<Exception, Directive> _decider;
-        private readonly bool _loggingEnabled;
+        public int MaxNumberOfRetries { get; private set; }
+        public int WithinTimeRangeMilliseconds { get; private set; }
+        public IDecider Decider { get; private set; }
 
```

Comment:
Oh yes it will, if you expose them as Get-properties and have a constructor with arguments that has the same name as the properties. This works:
``` C#
public class Test
{
	private readonly int _nummer;
	private readonly string _sträng;

	public Test(int nummer, string sträng)
	{
		_nummer = nummer;
		_sträng = sträng;
	}
	public int Nummer { get { return _nummer; } }
	public string Sträng { get { return _sträng; } }
}

var test = new Test(4711, "Håkan");
var json = JsonConvert.SerializeObject(test);
var test2 = JsonConvert.DeserializeObject<Test>(json);
System.Diagnostics.Debug.Assert(test.Nummer == test2.Nummer);
System.Diagnostics.Debug.Assert(test.Sträng == test2.Sträng);
```